_source/_posts/2021-12-18-native-java-helidon.adoc

```diff
@@ -0,0 +1,403 @@
+---
+layout: blog_post
+title: "Build REST APIs and Native Java Apps with Helidon"
+author: matt-raible
+by: advocate
+communities: [java]
+description: "Learn how to create REST APIs and native images with Helidon, a Java microservices framework from Oracle."
+tags: [native, java, graalvm, helidon]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+github: https://github.com/oktadev/native-java-examples
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+Project Helidon is a set of Java libraries for writing microservices. It's an open source, Apache 2.0-licensed, Oracle-sponsored project and contains support for MicroProfile as well as a reactive, functional API. Helidon's two different style are known as Helidon MP and SE, respectively. The https://helidon.io[helidon.io] website says:
+
+====
+Since Helidon is simply a collection of Java libraries running on a fast Netty core, there is no extra overhead or bloat.
+====
+
+image::{% asset_path 'blog/native-java-helidon/helidon-mp.png' %}[alt=Helidon supports MicroProfile,width=400,align=center]
+
+Today, I'm going to put this statement to the test! First, I'll show you how to run a secure, OAuth 2.0-protected, Java REST API that allows JWT authentication. Then, I'll show you how to build a native image with GraalVM. Finally, I'll compare Helidon to the most popular Java REST API frameworks: Micronaut, Quarkus, and Spring Book.
+
+My link:/blog/2021/06/18/native-java-framework-comparison[Build Native Java Apps with Micronaut, Quarkus, and Spring Boot] post compares startup times and memory usage of these three frameworks. How does Helidon perform? Read on!
+
+**Prerequisites**
+
+- https://sdkman.io/[SDKMAN]
+- https://maven.apache.org[Maven]
+- https://httpie.io/[HTTPie] (a better version of cURL)
+- An https://developer.okta.com[Okta Developer] Account
+
+toc::[]
+
+== Get Started with Helidon
+
+I created a GitHub repository you can clone and run to get started with Helidon quickly.
+
+[source,shell]
+----
+git clone https://github.com/oktadev/native-java-examples.git
```

Comment:
Don't forget to merge your `helidon` branch 😄

Reply:
This will be merged, so no branch reference necessary.